app/scripts/components/common/blocks/block-map.tsx

```diff
@@ -166,26 +251,132 @@ function MapBlock(props: MapBlockProps) {
     setMapBasemapId(basemapId);
   }, [basemapId]);
 
+  const baseTimeDensity = baseDataLayer?.data?.timeseries.timeDensity;
+  const compareTimeDensity = compareDataLayer?.data?.timeseries.timeDensity;
+
+  const compareToDate = useMemo(() => {
+    const theDate = selectedCompareDatetime ?? selectedDatetime;
+    return theDate instanceof Date && !isNaN(theDate.getTime())
+      ? theDate
+      : null;
+  }, [selectedCompareDatetime, selectedDatetime]);
+
+  const computedCompareLabel = useMemo(() => {
+    // Use a provided label if it exist.
+    const providedLabel = compareLabel ?? compareDataLayer?.data?.mapLabel;
+    if (providedLabel) return providedLabel as string;
+
+    // Default to date comparison.
+    return selectedDatetime && compareToDate
+      ? formatCompareDate(
+          selectedDatetime,
+          compareToDate,
+          baseTimeDensity,
+          compareTimeDensity
+        )
+      : null;
+  }, [
+    compareLabel,
+    compareDataLayer?.data?.mapLabel,
+    selectedDatetime,
+    compareToDate,
+    baseTimeDensity,
+    compareTimeDensity
+  ]);
+
+  const initialPosition = useMemo(() => center ? { lng: center[0], lat: center[1], zoom } : undefined , [center, zoom]);
+
   return (
     <Carto>
-      <MapboxMap
-        id={generatedId}
-        datasetId={datasetId}
-        layerId={layerId}
-        date={selectedDatetime}
-        isComparing={!!selectedCompareDatetime}
-        compareDate={selectedCompareDatetime}
-        compareLabel={compareLabel}
-        initialPosition={
-          center ? { lng: center[0], lat: center[1], zoom } : undefined
+      <Map id={generatedId} mapOptions={{...mapOptions, ...getMapPositionOptions(initialPosition)}} enableDefaultAttribution={false}>
+        <Basemap
+          basemapStyleId={mapBasemapId}
+        />
+        {
+          dataset &&
+          selectedDatetime &&
+          layerId &&
+          baseDataLayer?.data &&
+          (
+            <Layer
+              key={baseDataLayer.data.id}
+              id={`base-${baseDataLayer.data.id}`}
+              dataset={(baseDataLayer as unknown) as TimelineDatasetSuccess}
+              selectedDay={selectedDatetime}
```

Comment:
📓 : Casting because type is expecting `TimelineDatasetSuccess` but variable is only partial. We should probably update the type that Layer expect here to accept partial